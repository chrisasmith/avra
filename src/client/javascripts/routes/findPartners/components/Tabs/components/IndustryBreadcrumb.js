import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router'
import styles from './styles/IndustryBreadCrumb.scss'
import { buildRoute, routes } from '../../../../../routes'

const propTypes = {
  industry: PropTypes.object,
  params: PropTypes.object,
}

const IndustryBreadcrumb = ({ industry, params }) => {
  if (!industry) { return null }


  return (
    <div className={styles.breadcrumb}>
      <div>
        <h5>{industry.name}</h5>
      </div>
      {industry.subsections && industry.subsections.sort((a, b) => a.name.localeCompare(b.name)).map((subsection, idx) => (
        <div className={styles.breadCrumbLink} key={subsection.id}>
          {params.subsection_id === subsection.id || (!params.subsection_id && idx === 0) ? (
            subsection.name
          ) : (
            <Link to={buildRoute(routes.FINDPARTNERS.INDUSTRY_SUBSECTION, { industry_id: params.industry_id, subsection_id: subsection.id })}>{subsection.name}</Link>
          )}
        </div>
      ))}
    </div>
  )
}

IndustryBreadcrumb.propTypes = propTypes
export default IndustryBreadcrumb
