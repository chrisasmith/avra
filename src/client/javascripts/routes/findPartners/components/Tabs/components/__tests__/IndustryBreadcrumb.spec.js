import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import IndustryBreadcrumb from '../IndustryBreadcrumb'

describe('Industry Breadcrumb Component', () => {
  it('Should show', () => {
    const wrapper = shallow(<IndustryBreadcrumb
      industry={{ id: 'stuff', name: 'name' }}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Should show with types', () => {
    const wrapper = shallow(<IndustryBreadcrumb
      industry={{ id: 'stuff', name: 'name', subsections: [{ id: '1', name: 'first' }, { id: '2', name: 'second' }, { id: '3', name: 'a' }] }}
      params={{ subsection_id: null }}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('Should show active type', () => {
    const wrapper = shallow(<IndustryBreadcrumb
      industry={{ id: 'stuff', name: 'name', subsections: [{ id: '1', name: 'first' }, { id: '2', name: 'second' }, { id: '3', name: 'a' }] }}
      params={{ subsection_id: '2' }}
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
