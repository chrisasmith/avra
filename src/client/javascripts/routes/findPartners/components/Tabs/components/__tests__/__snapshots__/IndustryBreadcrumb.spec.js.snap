// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Industry Breadcrumb Component Should show 1`] = `
<div
  className="breadcrumb"
>
  <div>
    <h5>
      name
    </h5>
  </div>
</div>
`;

exports[`Industry Breadcrumb Component Should show active type 1`] = `
<div
  className="breadcrumb"
>
  <div>
    <h5>
      name
    </h5>
  </div>
  <div
    className="breadCrumbLink"
    key="3"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/find_partners/industries/undefined/3"
    >
      a
    </Link>
  </div>
  <div
    className="breadCrumbLink"
    key="1"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/find_partners/industries/undefined/1"
    >
      first
    </Link>
  </div>
  <div
    className="breadCrumbLink"
    key="2"
  >
    second
  </div>
</div>
`;

exports[`Industry Breadcrumb Component Should show with types 1`] = `
<div
  className="breadcrumb"
>
  <div>
    <h5>
      name
    </h5>
  </div>
  <div
    className="breadCrumbLink"
    key="3"
  >
    a
  </div>
  <div
    className="breadCrumbLink"
    key="1"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/find_partners/industries/undefined/1"
    >
      first
    </Link>
  </div>
  <div
    className="breadCrumbLink"
    key="2"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/find_partners/industries/undefined/2"
    >
      second
    </Link>
  </div>
</div>
`;
